### DISCUSSION TITLE
How do I use dbus in a docker container?

### DISCUSSION THREAD

**Discussion by JesseRiemens** [mikebrady/shairport-sync#1650] [Q&A] [answered]
I created my docker container using this command: `docker run -d --restart unless-stopped --net host --device /dev/snd \
    mikebrady/shairport-sync`

How can I use dbus to connect to it?

**Comment by mikebrady** [comment]
You may find some guidance in the issues, including closed issues, but your question is more of a Linux/Docker/D-Bus question…

**Reply by JesseRiemens** [reply]
Would you be interested in adding an example app into the repo that uses dart/phython? If I have free time I possibly can make something like that

**Comment by dubo-dubon-duponey** [comment, marked as answer]
@JesseRiemens 

The following example does disable convolution for eg:

```
docker run --name myshairport -d --restart unless-stopped --net host --device /dev/snd \ mikebrady/shairport-sync
docker exec -ti myshairport \
   dbus-send --print-reply --system \
  --dest=org.gnome.ShairportSync /org/gnome/ShairportSync org.freedesktop.DBus.Properties.Set string:org.gnome.ShairportSync \
  string:Convolution variant:boolean:false
```

(dbus-send obviously runs inside the container)

**Reply by dubo-dubon-duponey** [reply]
If you intend on accessing dbus programmatically instead, mounting the dbus socket from the host would be a start.

**Reply by JesseRiemens** [reply]
Thanks! In the meanwhile I've learned a lot about dbus and have been writing some apps for it. I'll for sure make it work like this. 

If I have time I'll PR some example app and docker-compose files for the newbie I was :)
